app/interactive_prompt.py

```diff
@@ -152,20 +153,29 @@ def focus(self):
         self.textBuffer.selectionAll()
 
     def formatCommand(self, cmdLine, lines):
-        formatter = {
+        formatters = {
             #".js": app.format_javascript.format
-            #".py": app.format_python.format
             #".html": app.format_html.format,
+            ".py": app.formatter.formatPython
         }
 
-        def noOp(data):
-            return data
-
         fileName, ext = os.path.splitext(self.view.host.textBuffer.fullPath)
+
         app.log.info(fileName, ext)
-        lines = self.view.host.textBuffer.doDataToLines(
-            formatter.get(ext,
-                          noOp)(self.view.host.textBuffer.doLinesToData(lines)))
+
+        formatter = formatters.get(ext)
+
+        if not formatter:
+            return lines, u'No formatter for extension {}'.format(ext)
+
+        text = self.view.host.textBuffer.doLinesToData(lines)
```

Comment:
Let's use ...textBuffer.linesToData(lines). I acknowledge that right now it's the same (it just calls through to doLinesToData).
Someday I plan to support editing binary files and at that point it may matter.